### PR TITLE
[TEST] Fix and cleanup TestKit codes

### PR DESCRIPTION
**Pull request by Arkatufus**
* Convert all async methods that returns an `IReadOnlyCollection` to `IAsyncEnumerable`
* Clean sync over async codes to always call `.ConfigureAwait(false)`
* Clean async await codes to always call  `.ConfigureAwait(false)`
* Fix erroneous HOCON strings
* Add missing `base.AfterAll()` in `AbstractSerializationTransportInformationSpec`, leaving a leaked ActorSystem living in the background.

### CODE REVIEW

File: src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Bugfix4360Spec.cs [hunk 1]
```diff
@@ -129,7 +129,7 @@ public void Should_recover_persistentactor_sqlite()
             recoveryActor.Tell("foo");
             recoveryActor.Tell("bar");
             recoveryActor.Tell(RecoverActor.DoSnapshot.Instance);
-            ExpectMsgAllOf("foo", "bar");
+            ExpectMsgAllOf(new []{ "foo", "bar" });
```
Comment (Aaronontheweb) [inline src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Bugfix4360Spec.cs:132]: Why is this necessary exactly? I thought the `params` handled this?

**Review by Aaronontheweb — COMMENTED** [review]
See a lot of the same change here, so I need to understand why it was made before I can keep reviewing this PR

File: src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Bugfix4360Spec.cs (hunk 1 above)
Comment (Arkatufus) [reply src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Bugfix4360Spec.cs:132]: `params` need to be replaced with `IEnumerable<T>` because we need to pass in a `CancellationToken` with a default value

Comment (Aaronontheweb) [reply src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Bugfix4360Spec.cs:132]: Got it, thank you. Are we going to warn users about that somehow? Roslyn rule?
